sdk/spring/spring-cloud-azure-stream-binder-eventhubs/pom.xml

```diff
@@ -106,6 +106,14 @@
           </rules>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.10.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
+        <configuration>
+          <failOnWarning>false</failOnWarning>
+        </configuration>
+      </plugin>
```

Comment:
This is to suppress what kind of warnings?

Reply:
[org.springframework.util.concurrent.ListenableFutureCallback](https://github.com/spring-projects/spring-framework/blob/9eaae0fe0450e0743f329feaaa3e23a979277438/spring-core/src/main/java/org/springframework/util/concurrent/ListenableFutureCallback.java#L32)

Reply:
Could you point me to the source code where [ListenableFutureCallback](https://github.com/spring-projects/spring-framework/blob/9eaae0fe0450e0743f329feaaa3e23a979277438/spring-core/src/main/java/org/springframework/util/concurrent/ListenableFutureCallback.java#L32) is used?

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/5bc550c9a5de4f8ee93a5b3141500ee34be3850d/sdk/spring/spring-integration-azure-core/src/main/java/com/azure/spring/integration/core/implementation/instrumentation/InstrumentationSendCallback.java#L13

https://github.com/Azure/azure-sdk-for-java/blob/968ec2ef4db334cba0fc71b2792a01a5a8d0c5b4/sdk/spring/spring-integration-azure-core/src/main/java/com/azure/spring/integration/core/handler/DefaultMessageHandler.java#L51

Reply:
We should change to `CompletableFuture` then, instead of using the deprecated one.